File: FactoryDesignPattern/Shapes_Example/Circle.java
package FactoryDesignPattern.Shapes_Example;

public class Circle extends Shape {
  private double circleRadius;

  public Circle(double radius) {
    this.circleRadius = radius;
  }

  @Override
  protected double area() {
    return Math.PI * this.circleRadius * this.circleRadius;
  }

  @Override
  public void describe() {
    System.out.printf("This is a Circle with area: %.2f%n", area());
  }
}

File: FactoryDesignPattern/Shapes_Example_Python/Shapes/Circle.py
import math
from Shapes.Shape import Shape

class Circle(Shape):

  def __init__(self, radius: float):
    self.radius = radius

  def area(self) -> float:
    return math.pi * self.radius * self.radius
  
  def describe(self) -> None:
    print(f"This is Circle with Radius: {self.radius} with Area: {self.area()}.")